Convert pkg/cmd/load_plugins.go to slog

ritoban23 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Make sure to read the Contributing Guide before submitting your PR: https://github.com/helm/helm/blob/main/CONTRIBUTING.md
2. If this PR closes another issue, add 'closes #<issue number>' somewhere in the PR summary. GitHub will automatically close that issue when this PR gets merged. Alternatively, adding 'refs #<issue number>' will not close the issue, but help provide the reviewer more context.-->

**What this PR does / why we need it**:

Converts `pkg/cmd/load_plugins.go` to use Go's structured logging package `log/slog` instead of the standard `log` package. This improves logging consistency, enables structured logs, and aligns with Helm's modernization efforts.

Closes #31454

**Special notes for your reviewer**:

- Only internal logging is affected; no user-facing changes.
- All related tests pass.

**If applicable**:
- [ ] this PR contains user facing changes (the `docs needed` label should be applied if so)
- [ ] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

ritoban23 committed:
Convert pkg/cmd/load_plugins.go to slog

- Replace log.Output() calls with slog.Debug() for structured logging
- Remove 'log' import and use 'log/slog' instead
- Removes the settings.Debug checks in favor of slog's built-in log level handling
- Improves logging consistency across the codebase

Fixes #13454

Signed-off-by: ritoban23 <ankudutt101@gmail.com>

benoittgt reviewed (approved):
~~LGTM~~

TerryHowe reviewed (changes requested):
Missing `fmt.Fprintf(os.Stderr, "failed to load plugins: %s\n", err)`

gjenkins8 reviewed (changes requested):
https://github.com/ritoban23/helm/blob/5a5674c36c44b2f8154fac2520cbf08817e4ef95/pkg/cmd/load_plugins.go#L66 (as @TerryHowe spotted) too please

robertsirc reviewed (dismissed):
LGTM

ritoban23 committed:
Convert remaining fmt.Fprintf to slog.Error in load_plugins.go

Address review feedback from TerryHowe and gjenkins8:
Convert fmt.Fprintf(os.Stderr, ...) on line 66 to slog.Error()
for consistent structured logging throughout the file.
